graphite/graphite.go

```diff
@@ -42,23 +60,45 @@ func (r *Request) Query(host string) (Response, error) {
 	if r.End != nil {
 		v.Add("until", fmt.Sprint(r.End.Unix()))
 	}
-	u := url.URL{
+	r.url = url.URL{
 		Scheme:   "http",
 		Host:     host,
 		Path:     "/render/",
 		RawQuery: v.Encode(),
 	}
-	resp, err := DefaultClient.Get(u.String())
+}
+
+func (r *Request) String() string {
+	return r.url.String()
+}
+
+func (r *Request) Query(host string) (Response, *Error) {
+	r.setUrl(host)
+	resp, err := DefaultClient.Get(r.url.String())
 	if err != nil {
-		return Response{}, err
+		e := NewError(fmt.Sprintf("Get(%s) failed", r.url.String()), "", err)
+		return Response{}, e
 	}
 	defer resp.Body.Close()
 	if resp.StatusCode != http.StatusOK {
-		return Response{}, errors.New(resp.Status)
+		body := make([]byte, 2000)
+		var msg string
+		_, err := resp.Body.Read(body)
+		if err == nil {
```

Comment:
that would mask the original error of the statuscode being bad. if reading body fails, i want to mention that as additional information, but we care about both errors, not just the last one.